### PR TITLE
BUG: frictionless API changes

### PR DESCRIPTION
**Pull request by lizgehret**
This PR fixes test failures caused by changes in the frictionless API, and new enforcements on their existing dataresource schema.

### CODE REVIEW

File: q2_stats/_transformer.py [hunk 1]
```diff
@@ -62,14 +62,18 @@ def _4(obj: pd.DataFrame) -> TabularDataResourceDirFmt:
 
         metadata['name'] = col
         metadata['type'] = schema_dtype
-        metadata_resource.schema.add_field(source=metadata)
 
-    metadata_resource.data = 'data.ndjson'
-    metadata_resource.format = 'ndjson'
+        metadata_obj.append(metadata)
+
+    metadata_dict = dict({'schema': metadata_obj})
```
Comment (Oddant1) [inline q2_stats/_transformer.py]: Casting this to dict is redundant

File: q2_stats/_transformer.py [hunk 2]
```diff
@@ -62,14 +62,18 @@ def _4(obj: pd.DataFrame) -> TabularDataResourceDirFmt:
...
+    metadata_dict['format'] = 'ndjson'
+    metadata_dict['path'] = 'data.ndjson'
+    json.dumps(metadata_dict)
 
     dir_fmt = TabularDataResourceDirFmt()
 
     dir_fmt.data.write_data(obj, pd.DataFrame)
-    metadata_resource.to_json(str(dir_fmt.path/'dataresource.json'))
+    with open(dir_fmt.path/'dataresource.json', 'w') as fh:
```
Comment (Oddant1) [inline q2_stats/_transformer.py]: Probably spaces around the / right? Since it's an operator? Idk though since lint isn't complaining.

**Review by Oddant1 — COMMENTED** [review]
Just nitpicky crap, should be fine

File: q2_stats/_transformer.py (hunk 2 above)
Comment (lizgehret) [reply q2_stats/_transformer.py]: Just tested this and it seems to work both ways 🤷‍♀️ @ebolyen do you have a preference?

File: q2_stats/_transformer.py [hunk 3]
```diff
@@ -62,14 +62,18 @@ def _4(obj: pd.DataFrame) -> TabularDataResourceDirFmt:
...
-        metadata_resource.schema.add_field(source=metadata)
 
-    metadata_resource.data = 'data.ndjson'
-    metadata_resource.format = 'ndjson'
+        metadata_obj.append(metadata)
+
+    metadata_dict = {'schema': metadata_obj}
+    metadata_dict['format'] = 'ndjson'
+    metadata_dict['path'] = 'data.ndjson'
+    json.dumps(metadata_dict)
```
Comment (ebolyen) [inline q2_stats/_transformer.py]: This poor JSON is yeeted into the void.
```suggestion
```

File: q2_stats/_transformer.py (hunk 2 above)
Comment (ebolyen) [reply q2_stats/_transformer.py]: I would agree on spaces, but if flake8 doesn't care, then I don't care either :-)

**Review by ebolyen — CHANGES_REQUESTED** [review]

**Comment by lizgehret** [comment]
Updated JSON dumps formatting, output looks like the below screenshot.

Screen Shot 2023-01-31 at 4 45 17 PM

**Review by ebolyen — APPROVED** [review]
Thanks @lizgehret!

**Pull request merged**